### ISSUE TITLE
mobx-state-router vs mobx-router

### ISSUE THREAD

**Issue by carlosagsmendes** [nareshbhatia/mobx-state-router#5]
Hi nareshbhatia,

Could you please further document the key differences between mobx-state-router and mobx-router?

I'm just starting with Mobx and would like to understand which one should I pick.

Thanks in advance!

**Comment by nareshbhatia** [comment]
@carlosagsmendes, the basic concepts are the same - both projects follow @mweststrate's guidance on how to handle state. I respect @kitze's work a lot. In fact, I had started with mobx-router first. Having said that, here's a list of improvements I have tried to make:

1. Cleaned up the API to make the underlying concepts crystal clear.
2. First class documentation, explaining what's going on under the scenes.
3. Written in TypeScript to define the interfaces unambiguously.
4. Provided the ability to fetch data asynchronously. Note that the router hooks all work with promises. The concept of "rejecting" a routing request has been thought through and documented.
5. Provided guidance for important use cases such as redirecting to a Sign In page and handling 404 errors.
6. Provided a [realistic app](https://mobx-shop.firebaseapp.com/) to show how the router is intended to be used. This app was carefully crafted to show best practices in MobX and state management. For example, see how [the stores are created](https://github.com/nareshbhatia/mobx-shop/blob/master/src/shared/stores/root.store.js) in a tree structure with the `RootStore` at the top. Also note how the transport layer/adapters are sent to the `RootStore` as a parameter instead of hard-coding in stores. This makes it easy to mock the transport layer for testing purposes.

Hope this helps. Glad to answer any follow-up questions.
